### PR TITLE
Refresh token

### PR DESCRIPTION
**Pull request by Korkkii**
Adds automatic refreshing of access token when it expires.

### CODE REVIEW

File: examples/access_token.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const express = require('express')
+const session = require('express-session')
+const gauth = require('../index.js')
+const app = express()
+const allowedLoginFromDomains = ['reaktor.fi', 'reaktor.com']
+// Initialize Google auth middleware. You need your Google app id and secret.
+const myGauth = gauth({
+  clientID: '384494927189-7ls0vh3mnsoc94s0q2dl5p21rdf9g82f.apps.googleusercontent.com',
+  clientSecret: 'C9YCedLkvkm_mml_L5Jizd-p',
```
Comment (pihvi) [inline examples/access_token.js]: Bit worrying to see secrets added in public commit..

File: package.json [hunk 2]
```diff
@@ -13,8 +13,10 @@
   },
   "homepage": "https://github.com/reaktor/express-gauth#readme",
   "dependencies": {
+    "moment": "^2.20.1",
```
Comment (rjoonas) [inline package.json]: I personally prefer [date-fns](https://date-fns.org/) over moment.

File: README.md [hunk 3]
```diff
@@ -90,12 +110,17 @@ gauth({
   // https://developers.google.com/identity/protocols/OpenIDConnect#authenticationuriparameters
   googleAuthorizationParams: {
     scope: ['profile', 'email'],
-    hostedDomain: 'reaktor.fi'
+    hostedDomain: 'reaktor.fi',
+    // To gain refreshable tokens
+    accessType: 'offline',
+    // This forces Google to send refresh tokens on each session start
+    // By default Google sends refresh token only once per authentication to Google
+    prompt: 'consent',
```
Comment (rjoonas) [inline README.md]: trailing comma

**Comment by Korkkii** [comment]
Added a forced relogin if trying to refresh access token when missing the refresh token. 

This can happen in a case where user starts a new session with a service, but user has previously authorized the service thus no consent authorization happens. Google sends the refresh token only on the first time authorizing a service, so in this case it won't send it. This forces a new consent screen which results in refresh token sent and added to user's session.

File: index.js [hunk 4]
```diff
@@ -44,7 +54,33 @@ module.exports = function expressGAuth(options) {
...
+              // Access token needs to be updated, but we are missing refresh
+              // token required to do it. Google provides refresh token only
+              // when authorizing the user through consent screen, so we
+              // reauthenticate the user and ask for their consent again in
+              // order to gain refresh token
+              const { googleAuthorizationParams } = config
+              const newGoogleParams = Object.assign(googleAuthorizationParams,
+                { prompt: 'consent' })
+              const newConfig = Object.assign(config,
+                { googleAuthorizationParams: newGoogleParams })
```
Comment (rjoonas) [inline index.js]: Why bind `newConfig` at all if the purpose is to just mutate `config`? Can't see `newConfig` being used anywhere.

**Comment by rjoonas** [comment]
I actually need this feature in another project so it would be good to get this in. Currently using this branch for local development and it seems to work.

**Comment by pihvi** [comment]
So make a release of this?

**Pull request merged**